IBM/ibm-crossplane-operator
# fix for reading data from configuration image

Issue: https://github.ibm.com/IBMPrivateCloud/roadmap/issues/51822
   fix in repo https://github.ibm.com/IBMPrivateCloud/bedrock-shim/pull/113
                https://github.com/IBM/ibm-common-service-operator/pull/738
Solution:
- [x] - create bedrock shim image with xpkg data in bedrock shim repo
- [X] - create configmap in operator
- [x] - attach data from bedrock-shim to configmap
- [x] - attach configmap to pod
- [x] - test pod if read confifuration from file
- [x] - test if init container tag was change to digest by CICD

**Should work with this Configuration CR**
we need to update it in Common Service installer 
```
apiVersion: pkg.ibm.crossplane.io/v1
kind: Configuration
metadata:
  labels:
    ibm-crossplane-provider: odlm
  name: ibm-crossplane-bedrock-shim-config
spec:
  ignoreCrossplaneConstraints: false
  package: ibm-crossplane-bedrock-shim-config
  packagePullPolicy: Never
  revisionActivationPolicy: Automatic
  revisionHistoryLimit: 1
  skipDependencyResolution: false

```

how to test before merge:
1. Prepare bedrock shim image with xpkg files inside image
2. Install Crossplane from strach -> Do not create CR Configuration 
3. Create Configmap , YAML from this PR
4. Update CSV as in this PR
5. Create Configuration CR with 
```
  package: ibm-crossplane-bedrock-shim-config
  packagePullPolicy: Never
```
6. Check if we can create KafkaClaim
